src/pipeline.rs

```diff
@@ -51,7 +52,29 @@ pub fn build_graphs(
 
     let mut line = String::new();
     while buffered.read_line(&mut line).unwrap() > 0 {
-        let split: Vec<&str> = line.trim().split('\t').collect();
+        let split: Vec<SmallVec<[String; SMALL_VECTOR_SIZE]>> = match &config.file_type {
+            FileType::JSON => {
+                let mut parsed = json::parse(&line).unwrap();
+                config.columns.iter().map({|c|
+                    if !c.complex {
+                        smallvec![parsed[&c.name].take_string().unwrap()]
+                    } else {
+                        parsed[&c.name].members().map(|v| {
+                            if v.is_string() {
+                                v.as_str().unwrap().to_owned()
+                            } else {
+                                v.dump()
+                            }
+                        }).collect()
+                    }
+                }).collect()
+            },
```

Comment:
IMO this code should be extracted to separate function for handling Json only. As this is for structured input data, I think that error handling should be introduced instead of .unwrap()

Reply:
Agreed that it's getting pretty bloated. I'll do some refactoring.